src/compile/compile-mustache.js

```diff
@@ -11,32 +13,44 @@ var Promise = require('bluebird');
  */
 function compile(options) {
 
-    return new Promise(function(resolve, reject) {
+    var templateName = getTemplateName(options.inputPath, options.useTemplateDirs);
+    var compiledTemplate = compileTemplate(options.code);
 
-        try {
+    return getOutputTemplate()
+        .then(function(outputTemplate) {
 
-            var templateName = getTemplateName(options.inputPath, options.useTemplateDirs);
+            var result = outputTemplate({
+                templateName: templateName,
+                compiledTemplate: compiledTemplate
+            });
 
-            var compiledTemplate = compileTemplate(options.code);
+            return {
+                code: result
+            };
 
-            var templateObject = '{ code: ' + compiledTemplate + ', partials: {}, subs: {} }';
+        });
 
-            var result = 'window["JST"] = window["JST"] || {};'

```

Comment:
Once this started to get more complex, concatenating strings of code just got super annoying. I'm now using Handlebars and a separate template file to generate the output.